### PR TITLE
Refactor User parse_orderby and add user_registered support

### PR DESCRIPTION
**Pull request by felipeelia**
### Description of the Change

This PR refactors User::parse_orderby(), adds support for user_registered into that and also handles lowercase `id` as a sort clause.

### Changelog Entry

Added: Support for ordering Users by `user_registered` and lowercase `id`.

### Credits

Props @felipeelia @burhandodhy

### CODE REVIEW

File: includes/classes/Indexable/User/User.php [hunk 1]
```diff
@@ -631,73 +634,72 @@ public function parse_orderby( $orderby, $default_order, $query_vars ) {
...
-						),
-					);
-				} elseif ( 'user_url' === $orderby_clause || 'url' === $orderby_clause ) {
-					$sort[] = array(
-						'user_url.raw' => array(
-							'order' => $order,
-						),
-					);
-				} elseif ( 'meta_value' === $orderby_clause ) {
+			if ( empty( $orderby_clause ) && 'rand' === $orderby_clause ) {
```
Comment (tott) [inline includes/classes/Indexable/User/User.php]: how would that second part ever be true

**Review by tott — CHANGES_REQUESTED** [review]
This seems wrong

File: includes/classes/Indexable/User/User.php (hunk 1 above)
Comment (felipeelia) [reply includes/classes/Indexable/User/User.php]: Totally true, @tott. I've changed that and also added a couple of other fields we don't offer support to.

**Review by tott — APPROVED** [review]
This looks good to me

**Pull request merged**